pkg/ccr/base/spec.go

```diff
@@ -358,6 +358,33 @@ func (s *Spec) GetAllViewsFromTable(tableName string) ([]string, error) {
 	return results, nil
 }
 
+func (s *Spec) RenameTable(destTableName string, renameTable *record.RenameTable) error {
+	// rename table may be 'rename table', 'rename rollup', 'rename partition'
+	var sql string
+	// ALTER TABLE table1 RENAME table2;
+	if renameTable.NewTableName != "" && renameTable.OldTableName != "" {
+		sql = fmt.Sprintf("ALTER TABLE %s RENAME %s", renameTable.OldTableName, renameTable.NewTableName)
+	}
+
+	// ALTER TABLE example_table RENAME ROLLUP rollup1 rollup2;
+	// if rename rollup, table name is unchanged
+	if renameTable.NewRollupName != "" && renameTable.OldRollupName != "" {
+		sql = fmt.Sprintf("ALTER TABLE %s RENAME ROLLUP %s %s", destTableName, renameTable.OldRollupName, renameTable.NewRollupName)
+	}
+
+	// ALTER TABLE example_table RENAME PARTITION p1 p2;
+	// if rename partition, table name is unchanged
+	if renameTable.NewParitionName != "" && renameTable.OldParitionName != "" {
+		sql = fmt.Sprintf("ALTER TABLE %s RENAME PARTITION %s %s;", destTableName, renameTable.OldParitionName, renameTable.NewParitionName)
+	}
+	if sql == "" {
+		return xerror.Errorf(xerror.Normal, "rename sql is empty")
+	}
```

Comment:
What happens if:
1. rename the table without the old name
2. modify distribution

![image](https://github.com/user-attachments/assets/fd38855c-3753-4761-8f37-ca86ee8759b1)

Reply:
1. Now the two situations will not generate binlog neither.
2. If the new name or old name is empty, ccr will return error